src/Scouting.elm

```diff
@@ -1,141 +1,198 @@
-                    ++ ", match - "
-                    ++ String.fromInt model.matchNum
-                    ++ ", station - "
-                    ++ model.driverStation
-            ]
-        ]
+module Scouting exposing (Model, Msg, driverStations, init, main, update, view)
+
+import Browser
+import Html
+import Html.Attributes exposing (placeholder, style, value)
+import Html.Events exposing (onClick, onInput)
+import List.Extra exposing (elemIndex, getAt)
+import Maybe exposing (withDefault)
+import Maybe.Extra exposing (unwrap)
+import Tuple exposing (first, second)
+
+
+main : Program () Model Msg
+main =
+    Browser.sandbox
+        { init = init
+        , view = view
+        , update = update
+        }
+
+
+type Msg
+    = NameInput String
+    | TeamInput String
+    | MatchInput String
+    | Start
+
+
+type alias Model =
+    { scouter : String
+    , team : Maybe Int
+    , match : Maybe Int
+    , driverStation : String
+    , isStarted : Bool
+    }
+
+
+driverStations : List (List Int)
+driverStations =
+    [ [ 1690, 1574, 3339, 254, 2056, 1323 ]
+    , [ 118, 1577, 1024, 2056, 1690, 254 ]
+    , [ 1574, 3339, 1577, 1323, 1024, 118 ]
+    , [ 3339, 1574, 1577, 1024, 118, 2056 ]
+    ]
+
+
+init : Model
+init =
+    { scouter = ""
+    , team = Nothing
+    , match = Nothing
+    , driverStation = ""
+    , isStarted = False
+    }
+
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        NameInput input ->
+            { model
+                | scouter = input
+            }
+
+        TeamInput input ->
+            { model
+                | team = String.toInt input
+                , driverStation = indexToName (String.toInt input) model.match
+            }
+
+        MatchInput input ->
+            { model
+                | match = String.toInt input
+                , driverStation = indexToName model.team (String.toInt input)
+            }
+
+        Start ->
+            { model
+                | isStarted = not model.isStarted
+                , driverStation = indexToName model.team model.match
+            }
+
+
+unwrapToString : Maybe Int -> String
+unwrapToString maybeInt =
+    unwrap "" String.fromInt maybeInt
+
+
+getMatchStations : Int -> List Int
+getMatchStations match =
+    -- == [team1, team2, team3, team4, team5, team6]
+    withDefault [ 0 ] <| getAt match driverStations
+
+
+stationIndex : Maybe Int -> Maybe Int -> String
+stationIndex team match =
+    -- team2 -> index = 2
+    unwrapToString << elemIndex (withDefault 0 team) <| getMatchStations (withDefault 0 match)
+
+
+indexToName : Maybe Int -> Maybe Int -> String
+indexToName team match =
+    case stationIndex team match of
+        "1" ->
+            "כחול 1"
+
+        "2" ->
+            "כחול 2"
+
+        "3" ->
+            "כחול 3"
+
+        "4" ->
+            "אדום 1"
+
+        "5" ->
+            "אדום 2"
+
+        "6" ->
+            "אדום 3"
+
+        _ ->
+            ""

```

Comment:
Hmmm... How do we make this case impossible?

Reply:
Make it return Just 'x' or Nothing?

Reply:
You're in the right direction